cwms-data-api/src/main/java/cwms/cda/data/dao/StreamReachDao.java

```diff
@@ -0,0 +1,236 @@
+                                    r.get(REACH_STREAM_ID_COLUMN, String.class),
+                                    r.get(REACH_UPSTREAM_LOCATION_COLUMN, String.class)),
+                            getBank(r.get(REACH_OFFICE_ID_COLUMN, String.class),
+                                    r.get(REACH_STREAM_ID_COLUMN, String.class),
+                                    r.get(REACH_DOWNSTREAM_LOCATION_COLUMN, String.class))))
+                    .collect(toList());
+        });
+    }
+
+    /**
+     * Retrieve a specific stream reach
+     * @param reachId - the id of the stream reach
+     * @param stationUnits - the station units used for stations
+     * @param officeId - the office id
+     * @return the stream reach
+     */
+    public StreamReach retrieveStreamReach( String officeId, String streamId, String reachId, String stationUnits) {
+        return retrieveStreamReaches(officeId, streamId, reachId, null, stationUnits).stream()
+                .findFirst()
+                .orElseThrow(() -> new NotFoundException("Stream Reach: " + reachId + " not found"));
+    }
+
+    /**
+     * Store a stream reach
+     * @param streamReach - the stream reach to store
+     * @param failIfExists - if true, fail if the stream reach already exists
+     */
+    public void storeStreamReach(StreamReach streamReach, boolean failIfExists) {
+        connectionResult(dsl, conn -> {
+            String failsIfExistsStr = OracleTypeMap.formatBool(failIfExists);
+            String ignoreNullsStr = OracleTypeMap.formatBool(true);
+            String downstreamLocId = streamReach.getDownstreamNode().getId().getName();
+            String upstreamLocId = streamReach.getUpstreamNode().getId().getName();
+            String configId = streamReach.getConfigurationId() == null ? null : streamReach.getConfigurationId().getName();
+            CWMS_STREAM_PACKAGE.call_STORE_STREAM_REACH(DSL.using(conn).configuration(), streamReach.getId().getName(), streamReach.getStreamId().getName(),
+                    failsIfExistsStr, ignoreNullsStr, upstreamLocId, downstreamLocId, configId, streamReach.getComment(),
+                    streamReach.getId().getOfficeId());
+            return null;
+        });
+    }
+
+    /**
+     * Rename a stream reach
+     * @param oldReachId - the old reach Id
+     * @param newReachId - the new reach Id
+     * @param officeId - the office Id
+     */
+    public void renameStreamReach(String officeId, String oldReachId, String newReachId) {
+        connection(dsl, conn -> {
+            CWMS_STREAM_PACKAGE.call_RENAME_STREAM_REACH(DSL.using(conn).configuration(), oldReachId, newReachId, officeId);
+        });
+    }
+
+    /**
+     * Delete a stream reach
+     * @param reachId - the id of the stream reach
+     * @param officeId - the office id
+     */
+    public void deleteStreamReach(String officeId, String reachId) {
+        connectionResult(dsl, conn -> {
```

Comment:
use `connection` instead of `connectionResult` instead of returning null

Reply:
updated to use `connection`